File: prueba-backend/src/main/java/com/empresa/proyecto/service/MaestraService.java
package com.empresa.proyecto.service;

import java.util.List;

import com.empresa.proyecto.dto.request.MaestraBuscar;
import com.empresa.proyecto.dto.response.Maestra;
import com.empresa.proyecto.dto.response.OutResponse;

public interface MaestraService {

	public OutResponse<List<Maestra>> listarMaestra(MaestraBuscar c);
}

File: prueba-frontend/src/app/modules/intranet/services/maestra.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { MaestraBuscar } from '../dto/request/maestra-buscar';
import { Maestra } from '../dto/response/maestra';
import { OutResponse } from '../dto/response/out.response';

@Injectable()
export class MaestraService {

  constructor(private http: HttpClient) { }

  public listarMaestra(req: MaestraBuscar): Observable<OutResponse<Maestra[]>> {
    return this.http.post<OutResponse<Maestra[]>>(`${environment.pruebaBackendUrl}/maestra/listarMaestra`, req);
  }
}
